runtime/interp.c

```diff
@@ -290,27 +322,28 @@ value caml_interprete(code_t prog, asize_t prog_size)
 #ifdef DEBUG
  next_instr:
   if (caml_icount-- == 0) caml_stop_here ();
-  CAMLassert(sp >= Caml_state->stack_low);
-  CAMLassert(sp <= Caml_state->stack_high);
+  CAMLassert(Stack_base(domain_state->current_stack) <= sp &&
+         sp <= Stack_high(domain_state->current_stack));
```

Comment:
Move it into its own `CAMLassert` as in trunk.

Reply:
Fixed in https://github.com/ocaml-multicore/ocaml/commit/eebbb5ab5a3be66817892b032949ca9387e4df4f